### DISCUSSION TITLE
non-linear heat conduction using ParNonlinearForm

### DISCUSSION THREAD

**Discussion by gokhalen** [mfem/mfem#3562] [General]
I'm trying to build a solve the non-linear heat conduction equation i.e. heat conduction equation in which the conductivity K depends on temperature. i.e. K = k0 + \alpha u  where u is the temperature. I'm basing my effort on [ex16p](https://github.com/mfem/mfem/blob/master/examples/ex16p.cpp) I want it to be as close as possible to the [nonlinear_elasticity_global_rom.cpp](https://github.com/LLNL/libROM/blob/master/examples/prom/nonlinear_elasticity_global_rom.cpp), I'd like to use a ParNonlinearForm for the term \div { \alpha u \grad u}.  Is the [ConvectionIntegrator](https://docs.mfem.org/html/classmfem_1_1ConvectionIntegrator.html) the right one to use for this purpose?

Thanks, 

-Nachiket

**Comment by vladotomov** [comment]
Hi Nachiket,

The `ParNonlinearForms` use `NonlinearFormIntegrators` that define the gradient calculations. You should take a look at example 10.

**Reply by gokhalen** [reply]
Thanks, this is not clear to me as of now. I need to think a bit about this.
